### PR TITLE
vcmi: fix week animation

### PR DESCRIPTION
**Pull request by rilian-la-te**
There was always day animation running on new weeks. Run week animation properly now.
Timer for week animation is estimation, maybe not accurately match H3.

### CODE REVIEW

File: client/adventureMap/CInfoBar.cpp [hunk 1]
```diff
@@ -99,6 +99,12 @@ std::string CInfoBar::VisibleDateInfo::getNewDayName()
 	}
 }
 
+int CInfoBar::VisibleDateInfo::getNewDayLength()
+{
+	// confirmed to match H3 for 3000 for NEWDAY
+	return std::max(3000, animation->getLength() * 2);
```
Comment (IvanSavenko) [inline client/adventureMap/CInfoBar.cpp:105]: Why this change?
Quick test shows that at least week2 & week3 are also visible for 3000 ms

Comment (rilian-la-te) [reply client/adventureMap/CInfoBar.cpp:105]: It is not fully visible, at least week 3 and 4. 1 and 2 - yes, visible, so this is why max.

Comment (rilian-la-te) [reply client/adventureMap/CInfoBar.cpp:105]: Week 3 and 4 requires more time to show full anim, but I unsure how many more.

Comment (IvanSavenko) [reply client/adventureMap/CInfoBar.cpp:105]: Ah, it looks like H3 always shows animation for 3 seconds, but instead calculates speed to play it in 1.5 seconds

Comment (rilian-la-te) [reply client/adventureMap/CInfoBar.cpp:105]: So, how I should change it to match H3?

Comment (rilian-la-te) [reply client/adventureMap/CInfoBar.cpp:105]: Replace 180 with 90? Or do it only on new weeks?

Comment (IvanSavenko) [reply client/adventureMap/CInfoBar.cpp:105]: Hmm.. All 5 files have different number of frames, but same duration

Maybe create method 
```
CShowableAnim::setDuration( int durationMilliseconds)
{
    frameTimeTotal  = durationMilliseconds/ (last - first)
}
```

And then use it as 
`animation->setDuration(1500)`

And revert changes in this method, back to "3000" constant

Comment (rilian-la-te) [reply client/adventureMap/CInfoBar.cpp]: done

**Review by IvanSavenko — APPROVED** [review]
